### PR TITLE
Adds TypeScript Types

### PR DESCRIPTION
**Pull request by bozdoz**
Hey cool project! Just trying it out and noticed it didn't have any types. Seemed simple enough to add them, based on the types given in the README. ✌️

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+declare module "boring-avatars" {
+  interface AvatarProps {
```
Comment (janmonschke) [inline index.d.ts:2]:
```suggestion
  export interface AvatarProps {
```

Exporting these makes it simpler to build custom components that wrap the Avatar components. Wdyt?

**Comment by janmonschke** [comment]
Great idea! I came here to do the same and was happy to see your PR 🥳

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    name?: string;
+    variant?: "marble" | "beam" | "pixel" | "sunset" | "ring" | "bauhaus";
+    colors?: string[];
+  }
+
+  interface AvatarComponent {
+    (props: AvatarProps, context?: any): React.ReactElement | null;
+  }
+
+  const Avatar: AvatarComponent
```
[the remark this answers is not in the document]
Comment (bozdoz) [reply index.d.ts:13]: `FunctionComponent` adds `children` to props, but this library does not use the `children` prop.  I copied the above from the `React.FunctionComponent` type and omitted what was unnecessary.

[the remark this answers is not in the document]
Comment (janmonschke) [reply index.d.ts:13]: Ah yes, good point  👍

**Comment by josepmartins** [comment]
Thank you @bozdoz, looks good to me 👍.

**Pull request merged**